bridge/contracts/BridgeRouter.sol

```diff
@@ -0,0 +1,239 @@
+            _factoryAddress()
+        );
+        uint256 implementationSize;
+        address implementation = _implementation;
+        assembly {
+            implementationSize := extcodesize(implementation)
+        }
+        require(
+            implementationSize > 0,
+            string(
+                abi.encodePacked(
+                    BridgeRouterErrorCodes.BRIDGEROUTER_UNEXISTENT_BRIDGEPOOL_IMPLEMENTATION_VERSION
+                )
+            )
+        );
+        address contractAddr = _deployStaticPool(bridgePoolSalt);
+        IBridgePool(contractAddr).initialize(ercContract_);
+        emit BridgePoolCreated(contractAddr);
+    }
+
+    /**
+     * @notice calculates salt for a BridgePool implementation contract based on tokenType and version
+     * @param tokenType_ type of token (1=ERC20, 2=ERC721)
+     * @param version_ version of the implementation
+     * @return calculated calculated salt
+     */
+    function getLocalERCImplementationSalt(uint8 tokenType_, uint16 version_)
+        public
+        pure
+        returns (bytes32)
+    {
+        string memory tag;
+        if (tokenType_ == 1) {
+            tag = "LocalERC20";
+        } else if (tokenType_ == 2) tag = "LocalERC721";
+        else tag = "Unknown";
+        return
+            keccak256(
+                bytes.concat(
+                    keccak256(abi.encodePacked(tag)),
+                    keccak256(abi.encodePacked(version_))
+                )
+            );
+    }
+
+    /**
+     * @notice calculates salt for a BridgePool contract based on ERC contract's address, tokenType, chainID and version_
+     * @param tokenContractAddr_ address of ERC contract of BridgePool
+     * @param tokenType_ type of token (1=ERC20, 2=ERC721)
+     * @param version_ version of the implementation
+     * @param chainID_ chain ID
+     * @return calculated calculated salt
+     */
+    function getBridgePoolSalt(
+        address tokenContractAddr_,
+        uint8 tokenType_,
+        uint256 chainID_,
+        uint16 version_
+    ) public pure returns (bytes32) {
+        return
+            keccak256(
+                bytes.concat(
+                    keccak256(abi.encodePacked(tokenContractAddr_)),
+                    keccak256(abi.encodePacked(tokenType_)),
+                    keccak256(abi.encodePacked(chainID_)),
+                    keccak256(abi.encodePacked(version_))
+                )
+            );
+    }
+
+    /**
+     * @notice creates a BridgePool contract with specific salt and bytecode returned by this contract fallback
+     * @param salt_ salt of the implementation contract
+     * @return contractAddr the address of the BridgePool
+     */
+    function _deployStaticPool(bytes32 salt_) internal returns (address contractAddr) {
+        uint256 contractSize;
+        assembly {
+            let ptr := mload(0x40)
+            mstore(ptr, shl(136, 0x5880818283335afa3d82833e3d82f3))
```

Comment:
Can we add more documentation about this smart contract? `0x5880818283335afa3d82833e3d82f3`

Reply:
Something like this:
```
   /*
      0x00    0x63         0x63XXXXXX  PUSH4 _code.length  size
      0x01    0x80         0x80        DUP1                size size
      0x02    0x60         0x600e      PUSH1 14            14 size size
      0x03    0x60         0x6000      PUSH1 00            0 14 size size
      0x04    0x39         0x39        CODECOPY            size
      0x05    0x60         0x6000      PUSH1 00            0 size
      0x06    0xf3         0xf3        RETURN
      <CODE>
```